performance_manager/lib/logging_utils.py

```diff
@@ -0,0 +1,70 @@
+import logging
+import time
+import uuid
+
+from typing import Union
+
+MdValues = Union[str, int, float]
+
+
+class ProcessLogger:
+    """
+    Class to help with logging events that happen inside of a function.
+    """
+
+    def __init__(self, process_name: str, **metadata: MdValues) -> None:
+        """
+        create a process logger with a name and optional metadata. a start time
+        and uuid will be created for timing and unique identification
+        """
+        self.process_name = process_name
+        self.uuid = uuid.uuid4()
+        self.start_time = time.time()
+        self.metadata = metadata
```

Comment:
Should this ( and ingestion logging) have some hard coded field to indicate it's part of performance manager?

Reply:
i think thats a good idea to do now. i would like both of the libraries to be pulling in shared lamp utils at some point, but we can do that at a later date.